pkg/controller/certificates/listers.go

```diff
@@ -29,8 +29,8 @@ import (
 // ListCertificateRequestsMatchingPredicates will list CertificateRequest
 // resources using the provided lister, optionally applying the given predicate
 // functions to filter the CertificateRequest resources returned.
-func ListCertificateRequestsMatchingPredicates(lister cmlisters.CertificateRequestNamespaceLister, selector labels.Selector, predicates ...predicate.Func) ([]*cmapi.CertificateRequest, error) {
```

Comment:
I see why in the tests but i think keeping it an interface should be nice

Reply:
Oooh yes you are right, changing to a func parameter instead of the existing interface feels... overkill 😅 I just have to mock this:

https://github.com/jetstack/cert-manager/blob/0bcf759a25544966e155ba67c41ea4f8a937238d/pkg/client/listers/certmanager/v1/certificaterequest.go#L62-L72

I guess I was feeling lazy implementing the mock functions `List` and `Get`... Hmm I guess taht's not much work in the end:

```go

type mockRequestLister struct {
	t                      *testing.T
	expectedListSelector   labels.Selector
	returnListCertRequests []*cmapi.CertificateRequest
	returnListErr          error
}

func (mock mockRequestLister) List(got labels.Selector) ([]*cmapi.CertificateRequest, error) {
	assert.Equal(mock.t, mock.expectedListSelector, got.String())
	return mock.returnListCertRequests, mock.returnListErr
}

func (mock mockRequestLister) Get(name string) (*cmapi.CertificateRequest, error) {
	mock.t.Error("Get is not expected to be called")
	return nil, nil
}
```

Is that what you had in mind?

Reply:
Done! Now using the lister interface to mock instead of a `list func()`